conanfile.py

```diff
@@ -135,6 +143,176 @@ def _python_installs(self):
 
         return python_installs
 
+    @staticmethod
+    def _is_repository_url(url):
+        # That will not work for ALL open-source projects, but should already get a large majority of them
+        return (url.startswith("https://github.com/") or url.startswith("https://gitlab.com/")) and "conan-center-index" not in url
+
+    def _make_pip_dependency_description(self, package, version, dependencies):
+        url = ["https://pypi.org/pypi", package]
+        if version is not None:
+            url.append(version)
+        url.append("json")
+
+        data = requests.get("/".join(url)).json()
+
+        dependency_description = {
+            "summary": data["info"]["summary"],
+            "version": data["info"]["version"],
+            "license": data["info"]["license"]
+        }
+
+        for url_data in data["urls"]:
+            if url_data["packagetype"] == "sdist":
+                sources_url = url_data["url"]
+                dependency_description["sources_url"] = sources_url
+
+                if not self.options.skip_licenses_download:
+                    # Download the sources to get the license file inside
+                    self.output.info(f"Retrieving license for {package}")
+                    response = requests.get(sources_url)
+                    response.raise_for_status()
+
+                    with tempfile.TemporaryDirectory() as temp_dir:
+                        sources_path = os.path.join(temp_dir, "sources.tar.gz")
+                        with open(sources_path, 'wb') as sources_file:
+                            sources_file.write(response.content)
+
+                        with tarfile.open(sources_path, 'r:gz') as sources_archive:
+                            license_file = "LICENSE"
+
+                            for source_file in sources_archive.getnames():
+                                if Path(source_file).name == license_file:
+                                    sources_archive.extract(source_file, temp_dir)
+
+                                    license_file_path = os.path.join(temp_dir, source_file)
+                                    with open(license_file_path, 'r') as file:
+                                        dependency_description["license_full"] = file.read()
```

Comment:
At this point we're like 9 indentations in, or so? 😅

Reply:
Yes indeed :disappointed: I'll make a separate function